src/Discovery/src/Consul/Registry/ConsulRegistration.cs

```diff
@@ -305,6 +299,12 @@ internal static AgentServiceCheck CreateCheck(int port, ConsulDiscoveryOptions o
             check.HTTP = uri.ToString();
         }
 
+        if (options.IsHeartBeatEnabled)
```

Comment:
So I looked at this issue a few months ago and did not get to a great resolution (at least in part because I'm not super familiar with how Consul expects these checks to work), but I have a local stash of the code that looks more like this for this line:
```
if (options.IsHeartBeatEnabled && options.HealthCheckUrl == null && options.HealthCheckPath == ConsulDiscoveryOptions.DEFAULT_HEALTH_PATH)
```

In hopes of a quicker turnaround to my questions than research, do you have any insights on how this feature is generally (outside of Steeltoe) expected to be configured? I don't know offhand if there is a generally accepted order of which check to use when, and it seems like there should be a set of rules _somewhere_ if there isn't an explicit setting for what type of check to use

Reply:
Hi @TimHess,
Thanks for your response!
To be perfectly honest with you, I don't have any insights on the order of checks... 
I noticed the issue and I was keen on providing a fix - but as I wrote the fix was very simple. 

I tried to get some inspiration from spring-cloud library but haven't found anything decisive.
Just docs and code:
- [Docs](https://github.com/spring-cloud/spring-cloud-consul/blob/dfbe776df8c7c3c4645c7c4a74f7e98c9109bb24/docs/modules/ROOT/partials/_configprops.adoc#L35)
- [More docs](https://github.com/spring-cloud/spring-cloud-consul/blob/dfbe776df8c7c3c4645c7c4a74f7e98c9109bb24/docs/modules/ROOT/pages/discovery.adoc#L186)
- [Code](https://github.com/spring-cloud/spring-cloud-consul/blob/dfbe776df8c7c3c4645c7c4a74f7e98c9109bb24/spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoRegistration.java#L218)

I've applied the change you mentioned you had stashed - it would do the trick as well.

Wondering then about the test for negative port then... It's only valid when we don't shortcircuit - so it requires to opt-out of the usage of heartbeat.

Reply:
The port is only relevant for an HTTP check. The TTL check requires the client to send a heartbeat signal to Consul (which records, but does not communicate with that port). The test for negative port seems correct to me

Reply:
Got it, thanks!